src/main/scala/com/databricks/spark/avro/AvroRelation.scala

```diff
@@ -74,6 +75,11 @@ case class AvroRelation(location: String)(@transient val sqlContext: SQLContext)
   private def newReader() = {
     val path = new Path(location)
     val fs = FileSystem.get(path.toUri, sqlContext.sparkContext.hadoopConfiguration)
+
+    if (fs.globStatus(path) == null) {
```

Comment:
Let's add a unit test for this. Also, globStatus may be expensive, so let's just reuse one call.